lib/iris/cube.py

```diff
@@ -1738,8 +1737,7 @@ def data(self):
                 result = as_concrete_data(self._dask_array,
                                           nans_replacement=ma.masked,
                                           result_dtype=self.dtype)
-                self._numpy_array = result
-                self.dtype = None
+                self.data = result
```

Comment:
Oh my ... this is interesting. I agree in the approach, but that'll now result in **both** the `dtype` and in addition the `fill_value` being cleared. It'll be interesting to see if this causes any of the existing tests to fail. I suspect something might barf.

Also, it means that we should join the dots here, and make sure that the `Cube.__init__` is setting the `data` by using the `self.data` setter ... and it's currently not! See [here](https://github.com/SciTools/iris/pull/2451/files#diff-d1f30240d0d146f09f58577e718ba6a6R719)